Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.21.6-alpine3.18@sha256:3354c3a94c3cf67cb37eb93a8e9474220b61a196b13c26f1c01715c301b22a69 AS build
+FROM golang:1.22.0-alpine@sha256:8e96e6cff6a388c2f70f5f662b64120941fcd7d4b89d62fec87520323a316bd9 AS build
```

Comment:
In addition to updating the go version, this also updates the alpine version. As of the time of this PR, Dependabot Version Update jobs for Docker won't handle updates across different OS versions such as -alpine3.18 and -alpine3.19 suffices. This is noted in the documentation here: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates#docker

There is a feature request for this functionality here:

- https://github.com/dependabot/dependabot-core/issues/4329